server/models/postgis/task.py

```diff
@@ -439,3 +435,37 @@ def as_dto(self):
         task_dto.task_history = task_history
 
         return task_dto
+
+    def as_dto_with_instructions(self, preferred_locale: str = 'en') -> TaskDTO:
+        """ Get dto with any task instructions """
+        task_dto = self.as_dto()
+
+        per_task_instructions = self.get_per_task_instructions(preferred_locale)
+
+        # If we don't have instructions in preferred locale try again for default locale
+        task_dto.per_task_instructions = per_task_instructions if per_task_instructions else self.get_per_task_instructions(
+            self.projects.default_locale)
+
+        return task_dto
+
+    def get_per_task_instructions(self, search_locale: str) -> str:
+        """ Gets any per task instructions attached to the project """
+        project_info = self.projects.project_info.all()
+
+        for info in project_info:
+            if info.locale == search_locale:
+                return self.format_per_task_instructions(info.per_task_instructions)
+
+    def format_per_task_instructions(self, instructions) -> str:
+        """ Format instructions by looking for X, Y, Z tokens and replacing them with the task values """
+        if not instructions:
+            return ''  # No instructions so return empty string
+
+        if not self.splittable and ('{x}/{y}/{z}' in instructions):
```

Comment:
No it's more performant to write AND statements like this.  All interpreters will evaluate the first part of the statement and if it's false not bother evaluating the second